components/DropdownLight/SubComponents/UseKeyPress.jsx

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+const useKeyPress = targetKey => {
+  const [keyPressed, setKeyPressed] = useState(false);
+
+  /* istanbul ignore next */
+  const downHandler = ({ key }) => {
+    if (key === targetKey) {
+      setKeyPressed(true);
+    }
+  };
+
+  /* istanbul ignore next */
+  const upHandler = ({ key }) => {
+    if (key === targetKey) {
+      setKeyPressed(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('keydown', downHandler);
+    window.addEventListener('keyup', upHandler);
+
+    return () => {
+      window.removeEventListener('keydown', downHandler);
+      window.removeEventListener('keyup', upHandler);
+    };
+  }, []);
+
+  return keyPressed;
+};
+
+export default useKeyPress;
```

Comment:
I think it's interesting to make a spike to think about ways to improve this hook's performance. Apparently every time the external component that uses it is rendered it creates a new listener for keyboard events and I believe this is a problem. Perhaps the solution is to implement a useCallback in the parent component.